odoo/addons/base/models/ir_cron.py

```diff
@@ -383,35 +484,29 @@ def _callback(self, cron_name, server_action_id, job_id):
 
             log_depth = (None if _logger.isEnabledFor(logging.DEBUG) else 1)
             odoo.netsvc.log(_logger, logging.DEBUG, 'cron.object.execute', (self._cr.dbname, self._uid, '*', cron_name, server_action_id), depth=log_depth)
-            _logger.info('Starting job `%s`.', cron_name)
+            _logger.info('Job `%s` (%s): Starting.', cron_name, self.id)
             start_time = time.time()
             self.env['ir.actions.server'].browse(server_action_id).run()
             self.env.flush_all()
             end_time = time.time()
-            _logger.info('Job done: `%s` (%.3fs).', cron_name, end_time - start_time)
+            _logger.info('Job `%s` (%s): Done in (%.3fs).', cron_name, self.id, end_time - start_time)
             if start_time and _logger.isEnabledFor(logging.DEBUG):
-                _logger.debug('%.3fs (cron %s, server action %d with uid %d)', end_time - start_time, cron_name, server_action_id, self.env.uid)
+                _logger.debug('Job `%s` (%s): Server action #%d with uid %d executed in (%.3fs).',
+                              cron_name, self.id, server_action_id, self.env.uid, end_time - start_time)
             self.pool.signal_changes()
-        except Exception as e:
+        except Exception:
             self.pool.reset_changes()
-            _logger.exception("Call from cron %s for server action #%s failed in Job #%s",
-                              cron_name, server_action_id, job_id)
-            self._handle_callback_exception(cron_name, server_action_id, job_id, e)
-
-    @api.model
-    def _handle_callback_exception(self, cron_name, server_action_id, job_id, job_exception):
-        """ Method called when an exception is raised by a job.
-
-        Simply logs the exception and rollback the transaction. """
-        self._cr.rollback()
+            _logger.exception('Job `%s` (%s): Server action #%s failed.', cron_name, self.id, server_action_id)
+            self._cr.rollback()
```

Comment:
Since we are at it, to be consistent with newer coding guidelines we have at the ORM
```suggestion
            self.env.cr.rollback()
```
Is this explicit rollback still needed?